src/js/tracks/text-track-settings.js

```diff
@@ -273,25 +242,184 @@ class TextTrackSettings extends Component {
+          {className: 'vjs-bg-color vjs-tracksetting'},
+          undefined,
+          [createEl('legend', {textContent: 'Background'})].concat(
+            this.createElSelect_('backgroundColor'),
+            createEl(
+              'span',
+              {className: 'vjs-bg-opacity vjs-opacity'},
+              undefined,
+              this.createElSelect_('backgroundOpacity')))),
+        createEl(
+          'fieldset',
+          {className: 'vjs-window-color vjs-tracksetting'},
+          undefined,
+          [createEl('legend', {textContent: 'Window'})].concat(
+            this.createElSelect_('windowColor'),
+            createEl(
+              'span',
+              {className: 'vjs-window-opacity vjs-opacity'},
+              undefined,
+              this.createElSelect_('windowOpacity'))))
+      ]);
+  }
+
+  /**
+   * Create font elements for the component
+   *
+   * @private
+   * @return {Element}
+   * @method createElFont_
+   */
+  createElFont_() {
+    return createEl(
+      'div',
+      {className: 'vjs-tracksettings-font'},
+      undefined,
+      [
+        createEl(
+          'div',
+          {className: 'vjs-font-percent vjs-tracksetting'},
+          undefined,
+          this.createElSelect_('fontPercent')),
+        createEl(
+          'div',
+          {className: 'vjs-edge-style vjs-tracksetting'},
+          undefined,
+          this.createElSelect_('edgeStyle')),
+        createEl(
+          'div',
+          {className: 'vjs-font-family vjs-tracksetting'},
+          undefined,
+          this.createElSelect_('fontFamily'))
+      ]);
+  }
+
+  /**
+   * Create controls for the component
+   *
+   * @private
+   * @return {Element}
+   * @method createElControls_
+   */
+  createElControls_() {
+    return createEl(
+      'div',
+      {className: 'vjs-tracksettings-controls'},
+      undefined,
+      [
+        createEl(
+          'button',
+          {className: 'vjs-default-button', textContent: 'Defaults'}),
+        createEl(
+          'button',
+          {className: 'vjs-done-button', textContent: 'Done'})
+      ]);
+  }
+
+  /**
    * Create the component's DOM element
    *
    * @return {Element}
    * @method createEl
    */
   createEl() {
-    const uniqueId = this.id_;
-    const dialogLabelId = 'TTsettingsDialogLabel-' + uniqueId;
-    const dialogDescriptionId = 'TTsettingsDialogDescription-' + uniqueId;
-
-    return super.createEl('div', {
-      className: 'vjs-caption-settings vjs-modal-overlay',
-      innerHTML: captionOptionsMenuTemplate(uniqueId, dialogLabelId, dialogDescriptionId),
-      tabIndex: -1
-    }, {
-      'role': 'dialog',
-      'aria-labelledby': dialogLabelId,
-      'aria-describedby': dialogDescriptionId
-    });
+    const settings = createEl(
+      'div',
+      {className: 'vjs-tracksettings'},
+      undefined,
+      [this.createElColors_(), this.createElFont_(), this.createElControls_()]);
+
+    const heading = createEl(
```

Comment:
might just be opinion here but I would like it if these blocks looked something like this: 
```JavaScript
const heading = createEl('div', {
  ...
  }, {
  ...
});
```